src/components/calculator.js

```diff
@@ -1,44 +1,64 @@
 /* eslint react/prefer-stateless-function: [0] */
 
 import React from 'react';
+import PropTypes from 'prop-types';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { result: 0 };
+    this.onClickCall = this.onClickCall.bind(this);
+  }
+
+  onClickCall = (e) => {
+    const { clickHandler } = this.props;
+    clickHandler(e.target.value);
   }
 
   render() {
-    const { result } = this.state;
+    const { total } = this.props;
+    const buttons = [
+      { style: '', value: 'AC' },
+      { style: '', value: '+/-' },
+      { style: '', value: '%' },
+      { style: 'operation', value: '÷' },
+      { style: '', value: '7' },
+      { style: '', value: '8' },
+      { style: '', value: '9' },
+      { style: 'operation', value: 'x' },
+      { style: '', value: '4' },
+      { style: '', value: '5' },
+      { style: '', value: '6' },
+      { style: 'operation', value: '-' },
+      { style: '', value: '1' },
+      { style: '', value: '2' },
+      { style: '', value: '3' },
+      { style: 'operation', value: '+' },
+      { style: 'zero', value: '0' },
+      { style: '', value: '.' },
+      { style: 'operation', value: '=' },
+    ];
+    const items = [];
+    buttons.forEach((button) => {
+      items.push(<button type="button" value={button.value} className={button.style} onClick={this.onClickCall}>{button.value}</button>);
+    });
     return (
       <div className="wrapper">
-        <div className="result">
-          {' '}
-          { result}
-          {' '}
-        </div>
-        <div> AC </div>
-        <div> +/- </div>
-        <div> % </div>
-        <div className="operation"> ÷ </div>
-        <div> 7 </div>
-        <div> 8 </div>
-        <div> 9 </div>
-        <div className="operation"> × </div>
-        <div> 4 </div>
-        <div> 5 </div>
-        <div> 6 </div>
-        <div className="operation"> - </div>
-        <div> 1 </div>
-        <div> 2 </div>
-        <div> 3 </div>
-        <div className="operation"> + </div>
-        <div className="zero"> 0 </div>
-        <div> . </div>
-        <div className="operation"> = </div>
+        <button className="result" type="button">
+          {total || 0}
+        </button>
```

Comment:
- [ ] The result not working as expected. Please fix it. As you can see in the video after ı click the operation ı can see the first number. I should see all numbers or operations in the result place after ı click.

OPTIONAL

https://user-images.githubusercontent.com/38635158/170789826-b89f967c-982f-4a64-8a5a-be8cdd1985e8.mp4



- [ ] It will be good if you fix the warning that showed in the console.